lib/host.js

```diff
@@ -63,11 +66,11 @@ Host.prototype.setDown = function() {
     //This host is not down, the pool is being shutdown
     return;
   }
-  this.log('warning', 'Setting host ' + this.address + ' as DOWN');
-  this.unhealthyAt = new Date().getTime();
   this.reconnectionDelay = this.reconnectionSchedule.next().value;
+  this.log('warning', util.format('Host %s considered as DOWN. Reconnection delay %dms.', this.address, this.reconnectionDelay));
+  this.unhealthyAt = new Date().getTime();
```

Comment:
So it looks like the problem was my guess that the reconnect was taking less than 1ms, reproduced this in the jenkins environment:

```
04:26:20 info HostConnectionPool Reconnection attempt to host 127.0.0.2:9042 failed { [Error: connect ECONNREFUSED 127.0.0.2:9042]
04:26:20   code: 'ECONNREFUSED',
04:26:20   errno: 'ECONNREFUSED',
04:26:20   syscall: 'connect',
04:26:20   address: '127.0.0.2',
04:26:20   port: 9042 }
04:26:20 Duration is less than or the same as reconnection delay! 1000 1000
04:26:20 << Duration is 1000 >>
04:26:20 Stop trying because we can't be considered as up.
```

A fix could be to either change the check in `hasTimePassed` to '>=' or to always force schedule a reconnect after a failed reconnect.